tests/test_parser.py

```diff
@@ -380,3 +454,46 @@ def test_tls_help(self):
         self.parser._set_defaults(None)
         help_output = self.parser._parser_tls.format_help()
         self.assert_snapshot('tls_help', help_output)
+
+
+class ParserModuleFunctionTestCase(unittest.TestCase):
+    # pylint: disable=protected-access
+    def test_create_parser(self):
+        description = 'parser description'
+        logfilename = 'logfilename'
+
+        parser = create_parser(description, logfilename)
+
+        self.assertTrue(isinstance(parser, CliParser))
+        self.assertEqual(parser._logfilename, logfilename)
+        self.assertEqual(parser._bootstrap_parser.description, description)
+
+    @mock.patch('gvmtools.parser.TLSConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_unix_socket_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test()
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_tls_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test('tls', TLSConnection)
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.TLSConnection')
```

Comment:
I don't get why mocking is necessary here at all and why it is necessary to mock the two unrelated connection classes